express-notes-api/index.js

```diff
@@ -0,0 +1,93 @@
+const express = require('express');
+const notesObject = require('./data.json');
+const fs = require('fs');
+const app = express();
+app.use(express.json());
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('listening on port 3000');
+});
+
+app.get('/api/notes', (req, res) => {
+  const notesArray = [];
+  for (const note in notesObject.notes) {
+    notesArray.push(notesObject.notes[note]);
+  }
+  res.status(200).json(notesArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const id = req.params.id;
+  if (parseInt(id) <= 0 || isNaN(id)) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+  } else if (notesObject.notes[id]) {
+    res.status(200).json(notesObject.notes[id]);
+  } else {
+    res.status(404).json({ error: `cannot find note with id ${id}` });
+  }
+});
```

Comment:
All non-integer ids should result in a `400 Bad Request` (ex: `GET /api/notes/564.23535`).